context_go17.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// +build go1.7
+
+package httprequest
+
+import (
+	"context"
+	"net/http"
+)
+
+func contextFromRequest(req *http.Request) (context.Context, *http.Request, context.CancelFunc) {
+	ctx := req.Context()
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Given that post-1.7 contexts are cancelled at the end of the request anyway, this seems unnecessary.